tests/gtest/avifjpeggainmaptest.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <math.h>
+
+#include <array>
+#include <tuple>
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace libavif {
+namespace {
+
+// Used to pass the data folder path to the GoogleTest suites.
+const char* data_path = nullptr;
+
+//------------------------------------------------------------------------------
+
+TEST(JpegTest, ReadJpegWithGainMap) {
+  for (const char* filename : {"paris_exif_xmp_gainmap_bigendian.jpg",
+                               "paris_exif_xmp_gainmap_littleendian.jpg"}) {
+    SCOPED_TRACE(filename);
+
+    const testutil::AvifImagePtr image =
+        testutil::ReadImage(data_path, filename, AVIF_PIXEL_FORMAT_YUV444, 8,
+                            AVIF_CHROMA_DOWNSAMPLING_AUTOMATIC,
+                            /*ignore_icc=*/false, /*ignore_exif=*/false,
+                            /*ignore_xmp=*/true, /*allow_changing_cicp=*/true,
+                            /*ignore_gain_map=*/false);
+    ASSERT_NE(image, nullptr);
+    ASSERT_NE(image->gainMap.image, nullptr);
+    EXPECT_EQ(image->gainMap.image->width, 512u);
+    EXPECT_EQ(image->gainMap.image->height, 384u);
+    EXPECT_EQ(image->xmp.size, 0u);
```

Comment:
Does this test require ignoring xmp? If so, a quick comment explaining why would be welcomed.